### PR TITLE
`v11.x.x` updates to `KeyboardKey`

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/29

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by johanstromqvist** [comment]
Please ping when ready for review again

**Comment by sophschneider** [comment]
@johanstromqvist This PR is ready for review now!

## Useful links
* [Design link](https://www.figma.com/file/hL9XRyHGqU8wUtLdOgs0vI/Polaris-Uplift-Library?type=design&node-id=7293-201736&t=yMams9DENPokkplL-0)
* [Storybook link](https://5d559397bae39100201eedc1-xjtxtpqxfm.chromatic.com/?path=/story/all-components-keyboardkey--small&globals=polarisSummerEditions2023:true)

## Large `KeyboardKey` variant converging with small one
@johanstromqvist  can we get confirmation that we are deprecating the default large `KeyboardKey` in favour of making the small one the standard? I couldn't find the decision written anywhere

| Before | After |
|-|-|
| Screenshot 2023-05-31 at 6 44 23 PM | Screenshot 2023-05-31 at 6 44 28 PM |

## Small `KeyboardKey` variant
| Before | After |
|-|-|
| Screenshot 2023-05-31 at 6 30 08 PM | Screenshot 2023-05-31 at 6 30 13 PM |

File: polaris-react/src/components/LegacyCard/LegacyCard.scss [hunk 1]
```diff
@@ -5,6 +5,14 @@
   box-shadow: var(--p-shadow-md);
   outline: var(--p-border-width-1) solid transparent;
 
+  #{$se23} & {
+    /* stylelint-disable -- SE23 temporary styles */
+    box-shadow: 0px 1px 0px rgba(0, 0, 0, 0.07), inset 0px -1px 0px #d4d4d4,
```
Comment (aveline) [inline polaris-react/src/components/LegacyCard/LegacyCard.scss]: Does this map to a shadow token now that can be used instead?

File: polaris-react/src/components/LegacyCard/LegacyCard.scss [hunk 2]
```diff
@@ -51,11 +67,20 @@
...
+      padding: var(--p-space-4);
+    }
   }
 
   + .Section {
     border-top: var(--p-border-width-1) solid var(--p-color-border-subdued);
 
+    #{$se23} & {
+      /* stylelint-disable-next-line polaris/color/function-disallowed-list -- se23 temporary styles */
+      border-top: 1px solid rgba(235, 235, 235, 1);
```
Comment (aveline) [inline polaris-react/src/components/LegacyCard/LegacyCard.scss]: Maybe this can be a border token now?

**Comment by johanstromqvist** [comment]
Thanks @sophschneider!

> @johanstromqvist can we get confirmation that we are deprecating the default large `KeyboardKey` in favour of making the small one the standard? I couldn't find the decision written anywhere

I'll look into it. How about we ship it and update it if that's not the case?

- I noticed that the component isn't included in the component TESTING library
- Also noticed that currently there appears to be 3 variants; omit size prop omitted (top), `small` (middle), and `large` (bottom). 

image

File: polaris-react/src/components/LegacyCard/LegacyCard.scss (hunk 2 above)
Comment (sophschneider) [reply polaris-react/src/components/LegacyCard/LegacyCard.scss]: Sorry @aveline, this file change must be from the branch being outdated. I rebased!

**Comment by sophschneider** [comment]
@johanstromqvist where did you find the `Large` variant? according the code and the [component docs](https://polaris.shopify.com/components/images-and-icons/keyboard-key) there's only:
* omit size prop
* small

**Review by johanstromqvist — APPROVED** [review]

**Comment by johanstromqvist** [comment]
> where did you find the Large variant?

@sophschneider I just tried it in code sandbox. Based on your message about "small" and "large" I just assumed it existed.

**Review by aveline — APPROVED** [review]

**Comment by sophschneider** [comment]
@johanstromqvist yeah that large state doesn't exist. It would give you a type error in typescript like this. Even though it looks different that is just an error state of the key

Screenshot 2023-06-01 at 11 21 45 AM

**Pull request merged**